pkg/wallet/account/account.go

```diff
@@ -0,0 +1,246 @@
+package account
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion    = 1
+	AccountUnknownVersion = 0
+)
+
+var (
+	ErrInvalidPrivateKey = errors.New("invalid private key")
+	ErrGeneratingKeyPair = errors.New("generating ed25519 key-pair")
+	ErrInvalidParameter  = errors.New("invalid parameter")
+)
+
+type Account struct {
+	Version      uint8                      `yaml:"Version"`
+	Nickname     string                     `yaml:"Nickname,omitempty"`
+	Address      *types.Address             `yaml:"Address,omitempty"`
+	Salt         [16]byte                   `yaml:"Salt,flow"`
+	Nonce        [12]byte                   `yaml:"Nonce,flow"`
+	CipheredData *types.EncryptedPrivateKey `yaml:"CipheredData,flow"`
+	PublicKey    *types.PublicKey           `yaml:"PublicKey,flow"`
+}
+
+func New(
+	version uint8,
+	nickname string,
+	address *types.Address,
+	salt [16]byte,
+	nonce [12]byte,
+	encryptedPrivateKey *types.EncryptedPrivateKey,
+	publicKey *types.PublicKey,
+) (*Account, error) {
+	if version > AccountLastVersion {
+		return nil, fmt.Errorf("%w: %d", ErrInvalidParameter, version)
```

Comment:
The error message will be "invalid parameter: 0". Can you change it to something easier to understand ?